CMakeLists.txt

```diff
@@ -33,6 +39,10 @@ elseif (${CMAKE_C_COMPILER_ID} STREQUAL MSVC)
     set(CMAKE_C_FLAGS "-W1")
     set(CMAKE_C_FLAGS_RELEASE "/O2 /Ot ")
     set(CMAKE_C_FLAGS_DEBUG "/Z7 /DEBUG")
+elseif (${CMAKE_C_COMPILER_ID} STREQUAL Intel)
+    set(CMAKE_C_FLAGS "-fPIC -Wall -qopenmp")
+    set(CMAKE_C_FLAGS_RELEASE "-O3")
+    set(CMAKE_C_FLAGS_DEBUG "-O0 -g")
 else ()
     message(FATAL_ERROR "We can't recongnise the C compiler!")
```

Comment:
Can we do some defaults if none of the above? e.g. PGI, or IBM, or Cray...

Reply:
Though this is not really an issue about this pull request...

Reply:
Hi Istvan,
No problem. I will add them soon. But I cannot test all flags at this moment, it will be great if you can point out the basic working flags for those compilers. In the next commit, we will have CXXFLAG and CFLAG to allow user flags for any purpose. 
Cheers,
Jianping